edelphi/src/main/java/fi/metatavu/edelphi/query/thesis/Multiple2DScalesThesisQueryPageHandler.java

```diff
@@ -0,0 +1,193 @@
+
+      selectedX.add(answerX != null ? answerX.getOption().getValue() : "NA");
+      selectedY.add(answerY != null ? answerY.getOption().getValue() : "NA");
+    }    
+    
+    addFragmentStringList(requiredFragment, "thesis", pageTheses);
+    addFragmentStringList(requiredFragment, "option", pageOptions);
+    addFragmentStringList(requiredFragment, "selectedX", selectedX);
+    addFragmentStringList(requiredFragment, "selectedY", selectedY);
+    
+    requiredFragment.addAttribute("labelX", pageLabelX);
+    requiredFragment.addAttribute("labelY", pageLabelY);
+    
+    addRequiredFragment(requestContext, requiredFragment);
+  }
+
+  @Override
+  protected void saveThesisAnswers(RequestContext requestContext, QueryPage queryPage, QueryReply queryReply) {
+    List<String> theses = getListOptionValue(queryPage, getDefinedOption(THESES_OPTION));
+    for (int thesisIndex = 0, thesisCount = theses.size(); thesisIndex < thesisCount; thesisIndex++) {
+      String fieldNameX = getFieldName(thesisIndex, "x");
+      String fieldNameY = getFieldName(thesisIndex, "y");
+      String fieldValueX = requestContext.getString(fieldNameX);
+      String fieldValueY = requestContext.getString(fieldNameY);
+      saveAnswer(requestContext, queryPage, queryReply, fieldNameX, fieldValueX);
+      saveAnswer(requestContext, queryPage, queryReply, fieldNameY, fieldValueY);
+    }    
+  }
+
+  @Override
+  protected void renderReport(PageRequestContext requestContext, QueryPage queryPage) {
+    // Live reports are not supported by multiple 2d scale pages
+  }
+
+  @Override
+  public void exportData(QueryExportContext exportContext) {
+    QueryFieldDAO queryFieldDAO = new QueryFieldDAO();
+    QueryQuestionCommentDAO queryQuestionCommentDAO = new QueryQuestionCommentDAO();
+    QueryQuestionOptionAnswerDAO queryQuestionOptionAnswerDAO = new QueryQuestionOptionAnswerDAO();
+
+    List<QueryReply> queryReplies = exportContext.getQueryReplies();
+    
+    QueryPage queryPage = exportContext.getQueryPage();
+    
+    boolean commentable = Boolean.TRUE.equals(this.getBooleanOptionValue(queryPage, getDefinedOption("thesis.commentable")));    
+    
+    List<String> theses = getListOptionValue(queryPage, getDefinedOption(THESES_OPTION));
+    for (int thesisIndex = 0, thesisCount = theses.size(); thesisIndex < thesisCount; thesisIndex++) {
+      String fieldNameX = getFieldName(thesisIndex, "x");
+      String fieldNameY = getFieldName(thesisIndex, "y");
+      
+      QueryOptionField queryFieldX = (QueryOptionField) queryFieldDAO.findByQueryPageAndName(queryPage, fieldNameX);
+      QueryOptionField queryFieldY = (QueryOptionField) queryFieldDAO.findByQueryPageAndName(queryPage, fieldNameY);
+
+      Messages messages = Messages.getInstance();
+      Locale locale = exportContext.getLocale();
+
+      int columnIndexX = exportContext.addColumn(String.format("%s/%s", queryPage.getTitle(), queryFieldX.getCaption()));
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define a constant instead of duplicating this literal "%s/%s" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1192)